packages/inventory-general-info/src/EditButton.js

```diff
@@ -5,33 +5,60 @@ import { usePermissions } from '@redhat-cloud-services/frontend-components-utili
 
 import { PencilAltIcon } from '@patternfly/react-icons';
 
-const EditButton = ({ link, onClick }) => {
+const InnerButton = ({ link, onClick }) => (
+    <a
+        className="ins-c-inventory__detail--action"
+        href={ `${window.location.href}/${link}` }
+        onClick={ onClick }
+    >
+        <PencilAltIcon />
+    </a>
+);
+
+InnerButton.propTypes = {
+    link: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired
+};
+
+let permissionsCache = undefined;
+
+const EditButtonUnknownPermissions = (props) => {
     const { hasAccess } = usePermissions('inventory', [
         'inventory:*:*',
         'inventory:hosts:write',
         'inventory:*:write'
     ]);
 
-    const canPerformActions = insights.chrome.isProd || hasAccess;
+    if (hasAccess) {
+        permissionsCache = hasAccess;
+    }
 
-    if (!canPerformActions) {
+    if (!hasAccess) {
```

Comment:
By default the `hasAccess` will have `undefined` might want to check `isLoading` from `useRbacHook` as well and perhaps show some loader/spinner.

Reply:
I hope the loading would be too fast to show anything. :smiley: 

Otherwise, all applications should probably set up the `writePermissions` prop, this should serve only as a fallback (probably it would be best to make the `writePermissions` required but right now I am not sure how many apps are using this component and how hard would be to get all updated).